packages/cosmic-swingset/tools/prometheus.js

```diff
@@ -0,0 +1,25 @@
+// https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format
+// metric_name [
+//   "{" label_name "=" `"` label_value `"` { "," label_name "=" `"` label_value `"` } [ "," ] "}"
+// ] value [ timestamp ]
+export const prometheusMetricNamePatt = '[a-zA-Z_:][a-zA-Z0-9_:]*';
+export const prometheusLabelNamePatt = '[a-zA-Z_][a-zA-Z0-9_]*';
+export const prometheusLabelPatt = String.raw`${prometheusLabelNamePatt}="(?:[^\\"\n]|\\(?:\\|"|n))*"`;
+export const prometheusSamplePatt = String.raw`^((${prometheusMetricNamePatt})(?:[{]${prometheusLabelPatt}(?:,${prometheusLabelPatt})*,?[}])?) +(\S+)( +-?[0-9]+|)$`;
+
+export const rLeadingPrometheusName = RegExp(
+  `^${prometheusMetricNamePatt}\\b`,
+  'gmu',
+);
+export const rPrometheusSample = RegExp(prometheusSamplePatt, 'gmu');
```

Comment:
I find the `r` prefix for an exported regexp does not convey the intent very well, and looks like a typo.  I'll defer to your judgement, but if you don't have a preference, how about:
```suggestion
export const leadingPrometheusNameRegExp = RegExp(
  `^${prometheusMetricNamePatt}\\b`,
  'gmu',
);
export const prometheusSampleRegExp = RegExp(prometheusSamplePatt, 'gmu');
```

?

I've also seen `const ALL_CAPS_NAMED_RE` elsewhere in the SDK to indicate they are a manifest constant regular expression.